src/writer/blocks/sharedblueprint.py

```diff
@@ -0,0 +1,98 @@
+"""
+Shared blueprint implementation for user-created reusable blueprints.
+
+Shared blueprints execute stored blueprint components by loading them from the component tree.
+The component's content.sourceBlueprintId field references the source blueprint.
+"""
+
+from typing import TYPE_CHECKING, Dict
+
+import writer.core_ui
+from writer.abstract import register_abstract_template
+from writer.blocks.base_block import BlueprintBlock
+from writer.ss_types import AbstractTemplate
+
+if TYPE_CHECKING:
+    from writer.blueprints import BlueprintRunner
+
+
+class SharedBlueprint(BlueprintBlock):
+    """
+    Shared blueprint that executes components from a source blueprint in the component tree.
+
+    The source blueprint is identified by the sourceBlueprintId content field.
+    """
+
+    @classmethod
+    def register(cls, type: str):
+        """Register the shared_blueprint type with its abstract template."""
+        super(SharedBlueprint, cls).register(type)
+        register_abstract_template(
+            type,
+            AbstractTemplate(
+                baseType="blueprints_node",
+                writer={
+                    "name": "Shared Blueprint",
+                    "description": "Executes a shared blueprint from the component tree.",
+                    "category": "Logic",
+                    "toolkit": "blueprints",
+                    "fields": {
+                        "payload": {
+                            "name": "Payload",
+                            "desc": "The value specified will be available using the template syntax, e.g. @{payload}.",
+                            "default": "{}",
+                            "type": "Text",
```

Comment:
I think you want `Object`, no?

```suggestion
                            "type": "Object",
```

Reply:
@madeindjs No, because the payload can either be a regular string, or a json payload inside a string `"{}"`. The same pattern is used for the Run Blueprint block